### DISCUSSION TITLE
Use `area:highway` for precise widths?

### DISCUSSION THREAD

**Discussion by svenpilz** [a-b-street/osm2streets#215] [General]
Hi fellow micro-mapper :).

I was wondering if there are plans for using `area:highway` if available to get the correct widths/outline of ways? There are already some good real world usages, for example in Berlin, with bigger crossings including pedestrian islands and y-junctions: https://www.openstreetmap.org/#map=18/52.46930/13.33266. The area around the linked crossing has also other examples, which are mostly connected, including street side parking and sidewalks: https://www.openstreetmap.org/#map=19/52.47142/13.33664.

Thanks and greetings from Berlin
Sven

**Comment by dabreegster** [comment]
Hi Sven! There is for sure appetite to use explicitly mapped `area:highway`s here, just a lack of time recently to work on osm2streets. Thanks for the pointers to good examples; these could be first test areas we try. Thinking about some simple steps to start integration...

- just extract `area:highway`s as a separate layer and toggle on/off in Street-Explorer, to manually inspect how much they do/don't match up with what the library infers now
- match up the areas with road and intersection objects we've got today
  - for roads, this should be easy, unless a road segment gets split multiple times between junctions
  - for intersections, it should be easy unless multiple nodes are involved. then maybe osm2streets is currently trying to merge them or not
- decide what to store
  - Right now, intersection polygons are explicitly stored, so this should be fine. But we'd need to also remember that the polygon is a "source of truth", and prevent a later transformation step from overwriting things
  - For roads, it's trickier; right now we're storing a corrected center-line and a specification of all lanes that sums up to a total width.
    - Maybe we need to begin storing an explicit optional polygon?
    - If we calculate a real width from the area, how do we distribute that width over all of the lanes in the road?
    - If the width varies along the road, how do we represent that?

@BudgieInWA, it's been a while... any thoughts about this?
